ibis/backends/impala/tests/test_client.py

```diff
@@ -266,11 +266,15 @@ def test_close_drops_temp_tables(con, test_data_dir):
 
     table = con.parquet_file(hdfs_path)
 
-    name = table.op().name
-    assert con.exists_table(name) is True
+    qualified_name = table.op().name
```

Comment:
After reading through this test, and the code path it's trying to exercise I no longer think this test is useful nor is there any real way to make this test reliable.

The main question here is: what should happen to any object that is referencing a temporary object from an *open* database connection, when that connection is closed?

I don't think this behavior is well defined, nor should we try to define it. However, since we're depending on `__del__` to clean up `ImpalaTemporaryTable`s (roughly speaking) when they are garbage collected, which should _definitely_ not be dropping tables when a connection is closed.

I will submit a PR to address that.

In this PR I think you can remove this test.

Reply:
Ok, so this is actually a different problem (though the lifetime problem still exists).

The issue is that `con.exists_table(name)` has different semantics than what you're inlining here.

`exists_table(like=name)` will return true if the list of tables matching `name` is non-empty.

What you've written here is to assert that whatever tables are listed contains `name` as an exact match.

The important thing to note is that this is only true when `con` is connected to the same database that `name` is contained in. Otherwise, the test will fail.

The solution is to write the assertion as `assert con.list_tables(like=name)`.